src/Database/Query.php

```diff
@@ -1413,13 +1414,14 @@ public function values($data)
      *
      * Can be combined with set() and where() methods to create update queries.
      *
-     * @param string|\Cake\Database\Expression\QueryExpression $table The table you want to update.
+     * @param string|\Cake\Database\ExpressionInterface $table The table you want to update.
      * @return $this
      */
     public function update($table)
     {
-        if (!is_string($table) && !($table instanceof QueryExpression)) {
-            $message = sprintf('Table must be of type string, got "%s"', gettype($table));
+        if (!is_string($table) && !($table instanceof ExpressionInterface)) {
+            $text = 'Table must be of type string or \Cake\Database\ExpressionInterface, got "%s"';
```

Comment:
You can use

    Table must be of type string or %s, got "%s"
    ...
    ExpressionInterface::class

right?